brayns/engine/Scene.cpp

```diff
@@ -164,19 +169,33 @@ size_t Scene::addModel(ModelDescriptorPtr modelDescriptor)
                 {true, true, modelDescriptor->getTransformation()});
     }
 
-    _computeBounds();
     return modelDescriptor->getModelID();
 }
 
 bool Scene::removeModel(const size_t id)
 {
     std::unique_lock<std::shared_timed_mutex> lock(_modelMutex);
-    auto model = _remove(_modelDescriptors, id, &ModelDescriptor::getModelID);
-    if (model)
+
+    if (supportsConcurrentSceneUpdates())
     {
-        model->callOnRemoved();
-        return true;
+        auto model =
+            _remove(_modelDescriptors, id, &ModelDescriptor::getModelID);
+        if (model)
+        {
+            model->callOnRemoved();
+            return true;
+        }
     }
+    else
+    {
+        auto model = _find(_modelDescriptors, id, &ModelDescriptor::getModelID);
+        if (model)
+        {
+            model->markForRemoval();
+            return true;
```

Comment:
missing `markModified()`, as similar to `addModel()`?

Reply:
It is done by the rockets plugin.